### PR TITLE
Docs: rename and spruce up boundary conditions section

### PR DESCRIPTION
**Pull request by eebasso**
Rename `PML.rst` to `boundary_conditions.rst` and fix citations and other references.

### CODE REVIEW

File: Docs/source/theory/boundary_conditions.rst [hunk 1]
```diff
@@ -0,0 +1,303 @@
+.. _theory-bc:
```
Comment (ax3l) [inline Docs/source/theory/boundary_conditions.rst:1]: Don't forget to update the table of contents with the new file name

File: Docs/source/theory/boundary_conditions.rst [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+Boundary conditions
```
Comment (ax3l) [inline Docs/source/theory/boundary_conditions.rst:3]: Title case?
```suggestion
Boundary Conditions
```

Comment (eebasso) [reply Docs/source/theory/boundary_conditions.rst:3]: Not sure. Seems like multiple case conventions are used. I personally prefer lower case for all words except the first one, proper names such as "Lorentz", and other such exceptions.

File: Docs/source/theory/boundary_conditions.rst (hunk 1 above)
Comment (ax3l) [reply Docs/source/theory/boundary_conditions.rst:1]: fixed

**Review by ax3l — APPROVED** [review]

File: Docs/source/theory/boundary_conditions.rst (hunk 2 above)
Comment (ax3l) [reply Docs/source/theory/boundary_conditions.rst:3]: I like title case, but I also miss the diverse spelling of words from German :D

**Pull request merged**